Tests/OptimizelyTests-DataModel/OptimizelyJSONTests.swift

```diff
@@ -0,0 +1,489 @@
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+import XCTest
+
+// MARK: - Sample Data and Setup
+
+class OptimizelyJSONTests: XCTestCase {
+    
+    private struct ValidSchema: Decodable, Equatable {
+        static func == (lhs: OptimizelyJSONTests.ValidSchema, rhs: OptimizelyJSONTests.ValidSchema) -> Bool {
+            return lhs.field1 == rhs.field1 && lhs.field2 == rhs.field2 && lhs.field3 == rhs.field3 &&
+                lhs.field4 == rhs.field4 && lhs.field5 == rhs.field5
+        }
+        
+        var field1: Int = 0
+        var field2: Double = 0.0
+        var field3: String = ""
+        var field4: Field4 = Field4()
+        var field5: Bool = false
+    }
+    
+    private struct Field4: Decodable, Equatable {
+        
+        static func == (lhs: OptimizelyJSONTests.Field4, rhs: OptimizelyJSONTests.Field4) -> Bool {
+            if (lhs.innerField1 != rhs.innerField1 || lhs.innerField2.count != rhs.innerField2.count) {
+                return false
+            }
+            for (index,value) in lhs.innerField2.enumerated() {
+                if rhs.innerField2[index] != value {
+                    return false
+                }
+            }
+            return true
+        }
+        
+        var innerField1: Int = 0
+        var innerField2: [InnerField2] = []
+        
+        enum CodingKeys: String, CodingKey {
+            case innerField1 = "inner_field1"
+            case innerField2 = "inner_field2"
+        }
+    }
+    
+    private enum InnerField2: Decodable, Equatable  {
+        case bool(Bool)
+        case double(Double)
+        case string(String)
+        
+        init(from decoder: Decoder) throws {
+            let container = try decoder.singleValueContainer()
+            if let x = try? container.decode(Bool.self) {
+                self = .bool(x)
+                return
+            }
+            if let x = try? container.decode(Double.self) {
+                self = .double(x)
+                return
+            }
+            if let x = try? container.decode(String.self) {
+                self = .string(x)
+                return
+            }
+            throw DecodingError.typeMismatch(InnerField2.self, DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "Wrong type for InnerField2"))
+        }
+    }
+    
+    private struct EmptySchema: Decodable {
+    }
+    
+    private struct IncorrectSchema: Decodable {
```

Comment:
I'll recommend add one more schema, that should be used directly with `getValue<T>` without encodable.